__tests__/app.test.js

```diff
@@ -398,4 +398,46 @@ describe('POST /api/articles/:article_id/comments', () => {
             })
         })
     })
+    it('responds with 404 if given an id that is valid but does not exist', () => {
+        const id = 456
+        const newComment = {
+            username: 'butter_bridge',
+            body: 'this is a new comment'
+        }
+        return request(app)
+        .post(`/api/articles/${id}/comments`)
+        .send(newComment)
+        .expect(404)
+        .then(({body}) => {
+            expect(body).toEqual({msg: 'route not found'})   
+        })
+    });
+    it('responds with 400 bad request when given an invalid ID', () => {
+        const id = 'bad'
+        const newComment = {
+            username: 'butter_bridge',
+            body: 'this is a new comment'
+        }
+        return request(app)
+        .post(`/api/articles/${id}/comments`)
+        .send(newComment)
+        .expect(400)
+        .then(({body}) => {
+            expect(body).toEqual({msg: 'bad request'})   
+        })
+    });
+    it('responds with a 404 user not found when username does not exist', () => {
+        const id = 3
+        const newComment = {
+            username: "Big_Lebowski",
+            body: "this is a new comment"
+        }
+        return request(app)
+        .post(`/api/articles/${id}/comments`)
+        .send(newComment)
+        .expect(404)
+        .then(({body}) => {
+            expect(body).toEqual({msg: 'user not found'})   
+        })
+    });
 })
```

Comment:
Just needs another test

- [ ]  400 missing required field from the post object

Reply:
🎉